worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,410 @@
+        spoiler_handle.write(f"3-2: {self.level_name_list[17]}\n")
+        spoiler_handle.write(f"3-3: {self.level_name_list[18]}\n")
+        spoiler_handle.write(f"3-4: {self.level_name_list[19]}\n")
+        spoiler_handle.write(f"3-5: {self.level_name_list[20]}\n")
+        spoiler_handle.write(f"3-6: {self.level_name_list[21]}\n")
+        spoiler_handle.write(f"3-7: {self.level_name_list[22]}\n")
+        spoiler_handle.write(f"3-8: {self.level_name_list[23]}\n")
+
+        spoiler_handle.write(f"\n4-1: {self.level_name_list[24]}\n")
+        spoiler_handle.write(f"4-2: {self.level_name_list[25]}\n")
+        spoiler_handle.write(f"4-3: {self.level_name_list[26]}\n")
+        spoiler_handle.write(f"4-4: {self.level_name_list[27]}\n")
+        spoiler_handle.write(f"4-5: {self.level_name_list[28]}\n")
+        spoiler_handle.write(f"4-6: {self.level_name_list[29]}\n")
+        spoiler_handle.write(f"4-7: {self.level_name_list[30]}\n")
+        spoiler_handle.write(f"4-8: {self.level_name_list[31]}\n")
+
+        spoiler_handle.write(f"\n5-1: {self.level_name_list[32]}\n")
+        spoiler_handle.write(f"5-2: {self.level_name_list[33]}\n")
+        spoiler_handle.write(f"5-3: {self.level_name_list[34]}\n")
+        spoiler_handle.write(f"5-4: {self.level_name_list[35]}\n")
+        spoiler_handle.write(f"5-5: {self.level_name_list[36]}\n")
+        spoiler_handle.write(f"5-6: {self.level_name_list[37]}\n")
+        spoiler_handle.write(f"5-7: {self.level_name_list[38]}\n")
+        spoiler_handle.write(f"5-8: {self.level_name_list[39]}\n")
+
+        spoiler_handle.write(f"\n6-1: {self.level_name_list[40]}\n")
+        spoiler_handle.write(f"6-2: {self.level_name_list[41]}\n")
+        spoiler_handle.write(f"6-3: {self.level_name_list[42]}\n")
+        spoiler_handle.write(f"6-4: {self.level_name_list[43]}\n")
+        spoiler_handle.write(f"6-5: {self.level_name_list[44]}\n")
+        spoiler_handle.write(f"6-6: {self.level_name_list[45]}\n")
+        spoiler_handle.write(f"6-7: {self.level_name_list[46]}\n")
+        spoiler_handle.write("6-8: King Bowser's Castle")
+
+    def create_item(self, name: str) -> Item:
+        data = item_table[name]
+
+        if data.useful:
+            classification = ItemClassification.useful
+        elif data.progression:
+            classification = ItemClassification.progression
+        elif data.trap:
+            classification = ItemClassification.trap
+        else:
+            classification = ItemClassification.filler
+
+        item = Item(name, classification, data.code, self.player)
+
+        return item
+
+    def create_regions(self):
+        create_regions(self.multiworld, self.player, get_locations
+                    (self.multiworld, self.player, self.boss_ap_loc, self.luigi_pieces, self),
+                    self.location_cache, self, self.boss_ap_loc, 
+                    self.level_location_list, self.luigi_pieces)
```

Comment:
```suggestion
    def create_regions(self):
        locations = get_locations(self)
        create_regions(self, locations)
```
And then rework create_regions and get_locations to just take those items.
Also would recommend using a different function name than create_regions here, but that's a soft recommendation, feel free to not do that.